### DISCUSSION TITLE
Throughput vs PC performance

### DISCUSSION THREAD

**Discussion by kwondaejang** [aligungr/UERANSIM#443] [Q&A]
Hi, Aligungr,

I have a question regarding throughput issue.

I got around downlink 100Mbps using free 5GC + UERANSIM.
=> CPU 2.3G x 8 (i9), RAM 16G

But I need more DL throughput, so I tried different PC and got downlink 250Mbps.
=> CPU 3.6G x 8 (i7), RAM 23.4G

So, maybe CPU and RAM performance affect throughput difference.

1. I saw somewhere you got 500Mbps, so can I ask your PC performance?
2. I am not using docker, but does that affect throughput?

Thank you.

**Comment by myonlystarWang** [comment]
@kwondaejang Hi, I can give you some of my experience.
I used the I5-10500 3.10Ghz(max to 4.5Ghz) 8GB RAM
It can easily reach 700-800Mbps.
I used the Open5gs + UERANSIM。

**Reply by kwondaejang** [reply]
@myonlystarWang 
Hi, Thank for sharing your experience.
After seeing your comment, I tried open5gs + UERANSIM with both above PC.

I am getting 200Mbps throughput for both PC. 

I am using below iperf cmd to test.
iperf3 -c 172.16.6.2 -i 1 -t 10

172.16.6.2 is another VM as a MEC, connected by internal interface with UPF.

May I ask how are you testing ?

**Reply by myonlystarWang** [reply]
Hi @kwondaejang 
My UERANSIM are two physical machines but not VM.I think that's where the problem is.You can try it.

**Reply by kwondaejang** [reply]
Thank you so much for your advice. 
I will try it later when I have some time:)

**Reply by hahada5** [reply]
@myonlystarWang  
Hi, thanks for your reply.
I still stuck with this problem.
I also use physical machines, one is running free5gc, and the other one is UERANSIM.
But I still get only 400Mbps with iperf3 tcp.( I expect to get at least 1Gbps.)
Did you have any suggestions for setting with UERANSIM, physical interface or anything else?
(My phsical machines are switch servers, performance is as great as yours and have larger memory.)

thank you very much!

iperf3 client result :
----------------------------------------------------------------------
Connecting to host 10.60.1.1, port 5201
[  4] local 60.60.0.1 port 41427 connected to 10.60.1.1 port 5201
[ ID] Interval           Transfer     Bandwidth       Retr  Cwnd
[  4]   0.00-1.00   sec  51.7 MBytes   433 Mbits/sec    0   2.41 MBytes
[  4]   1.00-2.00   sec  53.0 MBytes   444 Mbits/sec  738    807 KBytes
[  4]   2.00-3.00   sec  52.9 MBytes   444 Mbits/sec    0    862 KBytes
[  4]   3.00-4.00   sec  50.5 MBytes   424 Mbits/sec    0    898 KBytes
[  4]   4.00-5.00   sec  48.3 MBytes   405 Mbits/sec    0    921 KBytes
[  4]   5.00-6.00   sec  51.4 MBytes   431 Mbits/sec    0    960 KBytes
[  4]   6.00-7.00   sec  53.7 MBytes   451 Mbits/sec    0    999 KBytes
[  4]   7.00-8.00   sec  54.1 MBytes   454 Mbits/sec    0   1.01 MBytes
[  4]   8.00-9.00   sec  53.2 MBytes   446 Mbits/sec    0   1.05 MBytes
[  4]   9.00-10.00  sec  52.5 MBytes   440 Mbits/sec    0   1.08 MBytes
^C[  4]  10.00-10.51  sec  23.8 MBytes   388 Mbits/sec    0   1.10 MBytes
- - - - - - - - - - - - - - - - - - - - - - - - -
[ ID] Interval           Transfer     Bandwidth       Retr
[  4]   0.00-10.51  sec   545 MBytes   435 Mbits/sec  738             sender

**Reply by myonlystarWang** [reply]
Hi @hahada5，
    Could you please send your cpuinfo screenshot by command `lscpu`？
    In my experience, performance is very much related to the single CPU frequency.

**Comment by myonlystarWang** [comment]
Hi @hahada5，
    Could you please send your cpuinfo screenshot by command `lscpu`？
    In my experience, performance is very much related to the single CPU frequency.

**Reply by hahada5** [reply]
Hi @myonlystarWang 

my cpu frequency is 
CPU max MHz:         3800.0000
CPU min MHz:         1200.0000

thanks.

**Reply by myonlystarWang** [reply]
I don't know why, but the performance on this main frequency is so high
my cpu freq is 4200.00Mhz，the iperf3 result is up to 550-700Mbps

**Reply by hahada5** [reply]
It's okay. Thank you very much!
